src/main/java/org/bigredbands/mb/views/ProjectView.java

```diff
@@ -1428,10 +1429,12 @@ private void onClose() {
      * @param as If as==true, save as. If as==false, normal save.
      */
     private void onSave(boolean as) {
-        if (as) mainView.saveProjectAs();
-        else mainView.saveProject();
-        //TODO: Don't make saved true if the saving process is cancelled
-        saved = true;
+        try {
+            if (as) saved = mainView.saveProjectAs();
+            else saved = mainView.saveProject();
+        } catch (FileSelectionException e) {
+            mainView.displayError(e.getMessage());
```

Comment:
This is a nice touch - even though we shouldn't see errors in saving much (if at all), it will be good to report that to the user so they have at least a chance of diagnosing the problem.